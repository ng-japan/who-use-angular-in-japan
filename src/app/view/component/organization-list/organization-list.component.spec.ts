import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { generateOrganization } from '../../../testing/generator/generate-organization';
import { ToHtmlPipe } from '../../pipe/to-html.pipe';
import { OrganizationCardComponent } from '../organization-card/organization-card.component';
import { SafeHtmlPipe } from './../../pipe/safe-html.pipe';
import { OrganizationListComponent } from './organization-list.component';

describe('OrganizationListComponent', () => {
  let component: OrganizationListComponent;
  let fixture: ComponentFixture<OrganizationListComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [OrganizationListComponent, OrganizationCardComponent, SafeHtmlPipe, ToHtmlPipe],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(OrganizationListComponent);
    component = fixture.componentInstance;

    component.fetching = false;
    component.organizations = [generateOrganization(), generateOrganization(), generateOrganization()];

    fixture.detectChanges();
  });

  test('should be created', () => {
    expect(component).toBeTruthy();
  });

  test('matching snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
