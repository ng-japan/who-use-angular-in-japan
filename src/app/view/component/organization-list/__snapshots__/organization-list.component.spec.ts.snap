// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationListComponent matching snapshot 1`] = `
<app-organization-list
  fetching="false"
  organizations={[Function Array]}
>
  <div
    class="wrapper"
    fxlayout="column"
    fxlayoutalign="center center"
  >
    
    
    <div
      class="list"
      fxlayout="row wrap"
      fxlayout.lt-sm="column"
      fxlayoutalign="start stretch"
      fxlayoutgap="16px"
    >
      
      
      <app-organization-card
        class="listItem"
        fxflex="100%"
        fxflex.gt-lg="0 1 calc(25% - 16px)"
        fxflex.gt-md="0 1 calc(33.3% - 16px)"
        fxflex.gt-sm="0 1 calc(50% - 16px)"
      >
        <mat-card>
          <mat-card-title
            fxlayout="row"
            fxlayoutalign="start start"
            fxlayoutgap="8px"
          >
            <div
              class="mat-caption"
            >
              #1
            </div>
            <div
              class="mat-subtitle"
            >
              Dummy Company
            </div>
          </mat-card-title>
          <mat-card-subtitle>
            <a
              href="http://example.com"
              rel="noopener"
              target="_blank"
            >
              http://example.com
            </a>
          </mat-card-subtitle>
          <mat-card-content
            fxlayout="column"
            fxlayoutgap="16px"
          >
            <div
              class="mat-body-1 description"
            >
              <p>
                This company is dummy.
              </p>
              
  
            </div>
            <div
              fxlayooutalign="start center"
              fxlayout="row"
              fxlayoutgap="8px"
            >
              
              <img
                alt="angularjs icon"
                class="angularIcon"
                src="/assets/image/angularjs.svg"
              />
              
              <img
                alt="angular icon"
                class="angularIcon"
                src="/assets/image/angular.svg"
              />
              
            </div>
          </mat-card-content>
          <mat-card-actions>
            
          </mat-card-actions>
        </mat-card>
      </app-organization-card>
      
      <app-organization-card
        class="listItem"
        fxflex="100%"
        fxflex.gt-lg="0 1 calc(25% - 16px)"
        fxflex.gt-md="0 1 calc(33.3% - 16px)"
        fxflex.gt-sm="0 1 calc(50% - 16px)"
      >
        <mat-card>
          <mat-card-title
            fxlayout="row"
            fxlayoutalign="start start"
            fxlayoutgap="8px"
          >
            <div
              class="mat-caption"
            >
              #2
            </div>
            <div
              class="mat-subtitle"
            >
              Dummy Company
            </div>
          </mat-card-title>
          <mat-card-subtitle>
            <a
              href="http://example.com"
              rel="noopener"
              target="_blank"
            >
              http://example.com
            </a>
          </mat-card-subtitle>
          <mat-card-content
            fxlayout="column"
            fxlayoutgap="16px"
          >
            <div
              class="mat-body-1 description"
            >
              <p>
                This company is dummy.
              </p>
              
  
            </div>
            <div
              fxlayooutalign="start center"
              fxlayout="row"
              fxlayoutgap="8px"
            >
              
              <img
                alt="angularjs icon"
                class="angularIcon"
                src="/assets/image/angularjs.svg"
              />
              
              <img
                alt="angular icon"
                class="angularIcon"
                src="/assets/image/angular.svg"
              />
              
            </div>
          </mat-card-content>
          <mat-card-actions>
            
          </mat-card-actions>
        </mat-card>
      </app-organization-card>
      
      <app-organization-card
        class="listItem"
        fxflex="100%"
        fxflex.gt-lg="0 1 calc(25% - 16px)"
        fxflex.gt-md="0 1 calc(33.3% - 16px)"
        fxflex.gt-sm="0 1 calc(50% - 16px)"
      >
        <mat-card>
          <mat-card-title
            fxlayout="row"
            fxlayoutalign="start start"
            fxlayoutgap="8px"
          >
            <div
              class="mat-caption"
            >
              #3
            </div>
            <div
              class="mat-subtitle"
            >
              Dummy Company
            </div>
          </mat-card-title>
          <mat-card-subtitle>
            <a
              href="http://example.com"
              rel="noopener"
              target="_blank"
            >
              http://example.com
            </a>
          </mat-card-subtitle>
          <mat-card-content
            fxlayout="column"
            fxlayoutgap="16px"
          >
            <div
              class="mat-body-1 description"
            >
              <p>
                This company is dummy.
              </p>
              
  
            </div>
            <div
              fxlayooutalign="start center"
              fxlayout="row"
              fxlayoutgap="8px"
            >
              
              <img
                alt="angularjs icon"
                class="angularIcon"
                src="/assets/image/angularjs.svg"
              />
              
              <img
                alt="angular icon"
                class="angularIcon"
                src="/assets/image/angular.svg"
              />
              
            </div>
          </mat-card-content>
          <mat-card-actions>
            
          </mat-card-actions>
        </mat-card>
      </app-organization-card>
    </div>
    
  </div>
</app-organization-list>
`;
